amazon/dynamodb/client/src/main/kotlin/org/http4k/connect/amazon/dynamodb/mapper/DynamoDbQueryDsl.kt

```diff
@@ -36,13 +36,25 @@ class FilterExpression(
     val attributeValues: TokensToValues,
 )
 
-class DynamoDbScanAndQueryBuilder<HashKey : Any, SortKey : Any> {
+class DynamoDbScanAndQueryBuilder<HashKey : Any, SortKey : Any>(
+    private val hashKeyAttribute: Attribute<HashKey>,
+    private val sortKeyAttribute: Attribute<SortKey>?
+) {
+    object HashKeySubstitute
+    object SortKeySubstitute
 
     /**
      * See https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.KeyConditionExpressions.html
      */
     inner class KeyConditionBuilder {
 
+        val hashKey = HashKeySubstitute
+        val sortKey = SortKeySubstitute
```

Comment:
Note on the implementation: Actually, I would have preferred to simply write
```
object hashKey
object sortKey
```
here, but Kotlin doesn't allow objects in inner classes. That's why I moved the objects (with other names) one level "up" to `DynamoDbScanAndQueryBuilder` and defined the values `hashKey` and `sortKey` here instead. So `hashKey` and `sortKey` can only be used in the `keyCondition` closure, but not in `filterExpression`.

Reply:
Just discovered that it's not that simple. Without the values a user would need to import the objects. Having the values provides direct access.

Reply:
The anemic objects are gone ...